Fix #719 Invalid result for CBA_fnc_getNearest

nilsms opened the pull request:
Fix #719 Invalid result for CBA_fnc_getNearest
res = [[0,0,0], [[10,10,0],[1,1,0], [5,5,0]]] call CBA_fnc_getNearest;
result before fix => [10,10,0,[1,1,0],[5,5,0]]
result after fix => [1,1,0]

PabstMirror commented:
getNearest has 2 return types based on the presence of the radius param

So I think we want something like this
```
        if (count _this > 2) then {
            _return pushBack _x;
        } else {
            _radius = _distance;
            _return = _x;
        };
```
from https://github.com/nilsms/CBA_A3/blob/8d3519bf9e6ddf83a97a6a2247cb122ae96b4a2f/addons/common/fnc_getNearest.sqf

`isEqualType objNull` is the problem, because it doesn't work with arrayPositions as inputs

nilsms commented:
Your right, I missed the "or List of entities within given distance" part.

nilsms commented:
Quick rewrite of the patch to support returning multiple entities. Tested in game with both array positions and vehicles. Also added test case with radius.

Rewrote the branch commit history, old habit from other projects, not sure what you prefer.
